src/components/SettingsModal.js

```diff
@@ -0,0 +1,159 @@
+import { Box, Button, CircularProgress, FormControlLabel, Modal, Switch, TextField, Typography } from '@mui/material';
+import React, { useEffect, useRef, useState } from 'react';
+import { Controller, useForm } from 'react-hook-form';
+import { useDispatch, useSelector } from 'react-redux';
+import { componentProps, settingsClasses, settingsModalStyle } from 'utils/app.styleClasses';
+import ErrorMessage from './ErrorMessage';
+import { updateSettings } from 'api';
+import { updateInitialDeliveryFee } from 'redux/features/items/itemsSlice';
+
+const SettingsModal = ({ isModalOpen, setIsModalOpen }) => {
+  const { orderDeliveryFee, globalDeliveryDiscount, amountForFreeDelivery, percentageDiscount, settingsId } = useSelector((store) => store.items);
+  const [errorMessage, setErrorMessage] = useState(undefined);
+  const [isLoading, setIsLoading] = useState(false);
+  const submitFormRef = useRef(null);
+  const dispatch = useDispatch();
+
+  const {
+    control,
+    handleSubmit,
+    reset,
+    watch,
+    formState: { errors },
+  } = useForm();
+
+  let haveGlobalDeliveryDiscount = watch('haveGlobalDeliveryDiscount');
+  let deliveryFee = watch('deliveryFee');
+  let minAmountForFreeDelivery = watch('minAmountForFreeDelivery');
+  let discount = watch('discount');
+
+  let areValuesUnchanged =
+    deliveryFee === orderDeliveryFee && haveGlobalDeliveryDiscount === globalDeliveryDiscount && minAmountForFreeDelivery === amountForFreeDelivery && discount === percentageDiscount;
+
+  useEffect(() => {
+    if (!isLoading) {
+      let initialFormValues = { deliveryFee: orderDeliveryFee, haveGlobalDeliveryDiscount: globalDeliveryDiscount, minAmountForFreeDelivery: amountForFreeDelivery, discount: percentageDiscount };
+
+      // Sets the initial values to the form fields
+      reset(initialFormValues);
+    }
+  }, [isLoading]);
+
+  const handleSubmitSettings = async ({ deliveryFee, haveGlobalDeliveryDiscount, minAmountForFreeDelivery, discount }) => {
+    setErrorMessage(undefined);
+
+    if (areValuesUnchanged) {
+      // Prevents from making API call if theres no change
+      setIsModalOpen(false);
+      return;
+    }
+
+    setIsLoading(true);
+
+    const requestBody = {
```

Comment:
You don't need to use a watch in every input. Take a look [here](https://react-hook-form.com/get-started/#Registerfields)

Reply:
I am using watch in all of them because i want to check if the values of the input have been changed in order to stop the API call if theres no change. How would you do it otherwise?